ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -983,12 +977,7 @@ FEDataManager::spread(const int f_data_idx,
     // Accumulate data.
     f_data_ops->swapData(f_copy_data_idx, f_data_idx);
     f_data_ops->add(f_data_idx, f_data_idx, f_copy_data_idx);
-    for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
-    {
-        Pointer<PatchLevel<NDIM> > level = d_hierarchy->getPatchLevel(ln);
-        level->deallocatePatchData(f_copy_data_idx);
-    }
-    var_db->removePatchDataIndex(f_copy_data_idx);
+    d_cached_eulerian_data.restoreCachedPatchDataIndex(f_copy_data_idx);
```

Comment:
It would be nice if we could do this with RAII where we create check out and restore data by creating a struct. The destructor would update the cache. What do you think?

Reply:
@drwells, perhaps you could add this in a follow-up PR?

Reply:
(I agree that the current design is prone to what are effectively resource leaks. But the way we've been doing this using `VariableDatabase` is as well.)

Reply:
For reference, @boyceg and I had a conversation earlier about how this change is not trivial: e.g., such an object could not be copied.

Reply:
I just thought of a simple solution. Give me 10 minutes.

Reply:
Let me know what you think. I made a private inner `struct` that actually holds the data and handles deallocating it, and then a public inner `class` that holds a `std::shared_ptr` to the "real" data.

Reply:
:heavy_check_mark: